android/configure_java.md

```diff
@@ -0,0 +1,60 @@
+# Configure Java for Android
+If you are building Android via the command line (which includes our recommended pre-push hooks), you will need to configure Java. If not, you can just use Android Studio and do not need to read further.
+
+Notes:
+- this guide is written using macOS and will need to be updated to additionally support Linux and Windows.
+- if you find a better way to do configure Java, let us know!
+
+## Background
+At the time of writing, Android development seems to work best when using Java 8. By default `brew` and other package managers will install more recent versions that may throw errors during development.
+
+Caveat: Robolectric warns that Java 9 is needed for certain functionality. This seems inconsistent with the rest of the Android ecosystem, however.
+
+## Methods
+### Method #1: configure Java 8 from Android Studio
+Do the following:
+* Download and install Android Studio
+* Add the following line to your `~/.zshrc` or equivalent shell startup file: `export JAVA_HOME="/Applications/Android Studio.app/Contents/jre/jdk/Contents/Home"`
```

Comment:
Bonus points: Bryce is just about to make `JAVA_HOME` "win" in m-c `configure`, see https://phabricator.services.mozilla.com/D82418.